src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp

```diff
@@ -142,34 +184,54 @@ void ReplicatedMergeTreeTableMetadata::read(ReadBuffer & in)
 
     if (in.eof())
         data_format_version = 0;
-    else if (checkString("data format version: ", in))
+    else if (peekString("data format version: ", in))
```

Comment:
It's better to keep `checkString` if possible because peeking is too complex

Reply:
Then some must be extra careful when adding something new to metadata, e.g. two consecutive field name cannot have same prefix. I guess it's ok, don't expect to touch this code too often.